jni/tests/memory_test.cpp

```diff
@@ -0,0 +1,255 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+#include "faiss_wrapper.h"
+#include "nmslib_wrapper.h"
+
+#include <vector>
+#include <malloc.h>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "jni_util.h"
+#include "test_util.h"
+#include "faiss/utils/utils.h"
+
+using ::testing::NiceMock;
+using ::testing::Return;
+#define GTEST_COUT std::cerr << "[          ] [ INFO ]"
+
+TEST(FaissHNSWIndexMemoryTest, BasicAssertions) {
```

Comment:
can we add on all the tests what we are testing and what is the expectation as comments on top of all test function.

Also, I want to understand little bit here in terms of what are the failure scenario for these tests. may be you explained it in older comments, if yes can you point me there.

Reply:
@navneet1v 

These test just evaluate the memory only with EngineWrapper, not unit test.

When we want to introduce a new algorithm or engine, we prefer to evaluate the performance, memory, disk size. in benchmark tests, we can evaluate the performance as a single node. 

But we can not evaluate a engine/algorithm takes how much memory, because in benchmark jvm make it hard to evaluate the memory only in jni layer.

so i added these code and want to evaluate different algorithm/engine in different param, at index time, query time memory usage, and time usage.